tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/canonicalize/canonicalize.mlir

```diff
@@ -252,6 +252,28 @@ func.func @min_fold_float() -> tensor<6xf32> {
   func.return %2 : tensor<6xf32>
 }
 
+// CHECK-LABEL: clamp_scalar_fold
```

Comment:
Could you add a test with float that also covers NaN behavior?

Reply:
ok

Reply:
BTW, how to fold if 3.0 < x < 1.0 ? I think the result is NaN?

Reply:
XLA currently would compute min(3.0, max(x, 1.0)), so the folder would do the same thing as code generation does.